src/registrar/tests/test_models_domain.py

```diff
@@ -1,54 +1,487 @@
+"""
+Feature being tested: Registry Integration
+
+This file tests the various ways in which the registrar interacts with the registry.
+"""
 from django.test import TestCase
 from django.db.utils import IntegrityError
 
 from registrar.models import (
-    DomainApplication,
-    User,
     Domain,
 )
 from unittest import skip
 
 
-class TestDomain(TestCase):
-    def test_empty_create_fails(self):
+class TestDomainCreation(TestCase):
+    """Rule: An approved domain application must result in a domain"""
+
+    def setUp(self):
+        """
+        Background:
+            Given that a valid domain application exists
+        """
+        pass
+
+    @skip("not implemented yet")
+    def test_approved_application_creates_domain_locally(self):
+        """
+        Scenario: Analyst approves a domain application
+            When the DomainApplication transitions to approved
+            Then a Domain exists in the database with the same `name`
+            But a domain object does not exist in the registry
+        """
+        raise
+
+    @skip("not implemented yet")
+    def test_accessing_domain_properties_creates_domain_in_registry(self):
+        """
+        Scenario: A registrant checks the status of a newly approved domain
+            Given that no domain object exists in the registry
+            When `domain.is_active()` is called
+            Then Domain sends `commands.CreateDomain` to the registry
```

Comment:
I can see where this is causing confusion. `domain.is_active()` is merely an example of a property to access: this could be tested with any property.

`domain.is_active()` should really only result in "create domain" when the domain doesn't already exist in the registry (regardless of whether it is active -- although if it doesn't exist, of course it is not active)

Reply:
This makes so much more sense, just check that any given property being accessed causes the create. Thank you for clarifying.